### PR TITLE
Allow activating slider keyboard control in `Report` by clicking on linked carousel of images; and add a fix for Safari to allow for focussing of the slider

### PR DESCRIPTION
**Pull request by hoechenberger**
This works in Safari and Chrome now (once the slider is focused, I can use the left and right arrow keys to move through the figures):

https://github.com/mne-tools/mne-python/assets/2046265/3c398196-3e78-4e79-94e3-6a5bcb7eefb5

```python
# %%
from pathlib import Path

import matplotlib.pyplot as plt
import mne
import numpy as np
import scipy.ndimage

mne_logo_path = Path(mne.__file__).parent / "icons" / "mne_icon-cropped.png"
fig_array = plt.imread(mne_logo_path)
rotation_angles = np.linspace(start=0, stop=360, num=8, endpoint=False)

figs = []
captions = []
for angle in rotation_angles:
    # Rotate and remove some rounding errors to avoid Matplotlib warnings
    fig_array_rotated = scipy.ndimage.rotate(input=fig_array, angle=angle)
    fig_array_rotated = fig_array_rotated.clip(min=0, max=1)

    # Create the figure
    fig, ax = plt.subplots(figsize=(3, 3), layout="constrained")
    ax.imshow(fig_array_rotated)
    ax.set_axis_off()

    # Store figure and caption
    figs.append(fig)
    captions.append(f"Rotation angle: {round(angle, 1)}°")

report = mne.Report(title="Multiple figures example")
report.add_figure(fig=figs, title="Fun with figures! 🥳", caption=captions)
report.save("/tmp/report_custom_figures.html", overwrite=True)
```

Previously, keyboard controls didn't work in Safari as the slider would not receive focus after a click.

This PR fixes this and additionally adds a new functionality: clicking on the figure now also focuses the slider (in all browsers), activating keyboard controls.

### CODE REVIEW

File: mne/html_templates/report/slider.html.jinja [hunk 1]
```diff
@@ -24,14 +25,8 @@
...
-      <div id="corousel-{{ id }}" class="carousel carousel-dark" data-bs-interval="false" data-bs-wrap="false">
-        {# <!-- <div class="carousel-indicators">
-          {{for idx, caption in zip(range(len(images)), captions) }}
-          <button type="button" data-bs-target="#corousel-{{id}}" data-bs-slide-to="{{idx}}" class="active"
-            aria-current="true" aria-label="{{caption}}"></button>
-          {{endfor}}
-        </div> --> #}
-
+      <div id="corousel-{{ id }}" class="carousel carousel-dark" data-bs-interval="false" data-bs-wrap="false"
+        data-bs-keyboard="true">
```
Comment (hoechenberger) [inline mne/html_templates/report/slider.html.jinja:29]: just some cleanup

File: mne/html_templates/report/slider.html.jinja [hunk 2]
```diff
@@ -46,14 +41,14 @@
           {% endfor %}
         </div>
 
-        {# <!-- <button class="carousel-control-prev" type="button" data-bs-target="#corousel-{{id}}" data-bs-slide="prev">
+        {# <button class="carousel-control-prev" type="button" data-bs-target="#corousel-{{id}}" data-bs-slide="prev">
```
Comment (hoechenberger) [inline mne/html_templates/report/slider.html.jinja:44]: cleanup

File: mne/html_templates/report/slider.html.jinja [hunk 3]
```diff
@@ -46,14 +41,14 @@
...
           <span class="carousel-control-prev-icon" aria-hidden="true"></span>
           <span class="visually-hidden">Previous</span>
         </button>
         <button class="carousel-control-next" type="button" data-bs-target="#corousel-{{id}}" data-bs-slide="next">
           <span class="carousel-control-next-icon" aria-hidden="true"></span>
           <span class="visually-hidden">Next</span>
-        </button> --> #}
+        </button> #}
```
Comment (hoechenberger) [inline mne/html_templates/report/slider.html.jinja:51]: cleanup

File: mne/html_templates/report/slider.html.jinja [hunk 4]
```diff
@@ -1,4 +1,5 @@
-<div class="accordion-item slider {{ klass }}" id="{{ id }}" data-mne-tags="{% for tag in tags %} {{ tag }} {% endfor %}">
+<div class="accordion-item slider {{ klass }}" id="{{ id }}"
+  data-mne-tags="{% for tag in tags %} {{ tag }} {% endfor %}">
```
Comment (hoechenberger) [inline mne/html_templates/report/slider.html.jinja:2]: cleanup

**Review by larsoner — APPROVED** [review]
Can confirm it works great on Chrome at least, happy to trust your testing on Safari. Marking for merge-when-green and restarting unrelated failed CIs, thanks @hoechenberger !

**Comment by hoechenberger** [comment]
@larsoner I already tested on Safari, thanks for checking with Chrome!

**Pull request merged**